app/src/pages/inside/stepPage/modals/linkIssueModal/linkIssueModal.jsx

```diff
@@ -0,0 +1,263 @@
+import { activeProjectSelector } from 'controllers/user';
+import { externalSystemSelector } from 'controllers/project';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { showNotification, NOTIFICATION_TYPES } from 'controllers/notification';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { URLS } from 'common/urls';
+import { InputDropdown } from 'components/inputs/inputDropdown';
+import { GhostButton } from 'components/buttons/ghostButton';
+import { FormField } from 'components/fields/formField';
+import { Input } from 'components/inputs/input';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { validate, fetch } from 'common/utils';
+import styles from './linkIssueModal.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  linkButton: {
+    id: 'LinkIssueModal.linkButton',
+    defaultMessage: 'Link',
+  },
+  title: {
+    id: 'LinkIssueModal.title',
+    defaultMessage: 'Link issue',
+  },
+  addIssueIdTitle: {
+    id: 'LinkIssueModal.addIssueIdTitle',
+    defaultMessage: 'Add issue id',
+  },
+  addIssueButtonTitle: {
+    id: 'LinkIssueModal.addIssueButtonTitle',
+    defaultMessage: 'Add new issue',
+  },
+  issueLinkLabel: {
+    id: 'LinkIssueModal.issueLinkLabel',
+    defaultMessage: 'Link to issue',
+  },
+  issueIdLabel: {
+    id: 'LinkIssueModal.issueIdLabel',
+    defaultMessage: 'Issue ID',
+  },
+  issueLinkPlaceholder: {
+    id: 'LinkIssueModal.issueLinkPlaceholder',
+    defaultMessage: 'Enter link to issue',
+  },
+  projectInBtsTitle: {
+    id: 'LinkIssueModal.projectInBtsTitle',
+    defaultMessage: 'Project name in BTS',
+  },
+  closeModalWarning: {
+    id: 'LinkIssueModal.closeModalWarning',
+    defaultMessage: 'You have to save changes or cancel them before closing the window',
+  },
+  linkIssueSuccess: {
+    id: 'LinkIssueModal.linkIssueSuccess',
+    defaultMessage: 'Defect link successfully added',
+  },
+});
+
+@withModal('linkIssueModal')
+@injectIntl
+@reduxForm({
+  form: 'linkIssueForm',
+  validate: ({ issues }) => ({
+    issues:
+      issues &&
+      issues.map((issue) => ({
+        issueLink: (!issue.issueLink || !validate.url(issue.issueLink)) && 'urlHint',
+        issueId: (!issue.issueId || !validate.issueId(issue.issueId)) && 'issueIdHint',
+      })),
+  }),
+})
+@connect(
+  (state) => ({
+    url: URLS.testItemsAddIssues(activeProjectSelector(state)),
+    externalSystems: externalSystemSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+export class LinkIssueModal extends Component {
+  static propTypes = {
+    intl: PropTypes.object.isRequired,
+    url: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    externalSystems: PropTypes.array.isRequired,
+    initialize: PropTypes.func.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
```

Comment:
Maybe better to call this method `onChange`, because it's some callback function?

Reply:
It is a standart ReduxForm prop `change`.
https://redux-form.com/7.3.0/docs/api/props.md/#-code-change-field-string-value-any-function-code-

Reply:
Ok, I'm agree. Same naming is better.